deploy-gae-build-task/deploy-gae.ts

```diff
@@ -33,61 +24,65 @@ import Endpoint = exec.Endpoint;
+
   // Check that gcloud exists.
   const gcloudPath = task.which('gcloud', true);
   checkGcloudVersion(gcloudPath);
 
-  // Get inputs from GUI.
-  // The id of the GCP service endpoint to get the credentials from.
-  const endpointId = task.getInput('serviceEndpoint', true);
-  // The path of the deployment files.
-  const deploymentPath = task.getPathInput('deploymentPath', true);
-  // The name of the YAML file we want to run on.
-  const yamlFileName = task.getInput('yamlFileName', true);
-  // If true, copy the YAML file from the source folder to the deployment
-  // path.
-  const copyYaml = task.getBoolInput('copyYaml', true);
-  // The source folder the YAML file is in.
-  const sourceFolder = task.getPathInput('sourceFolder', copyYaml);
-  // The storage bucket to send to the --bucket parameter.
-  const storageBucket = task.getInput('storageBucket', false);
-  // The version to deploy.
-  const versionInput = task.getInput('version', false);
-  // Toggle between --promote and --no-promote parameters.
-  const promote = task.getBoolInput('promote', true);
-  // Toggle between --stop-previous-version and --no-stop-previous-version.
-  const stopPrevious = task.getBoolInput('stopPrevious', promote);
+  const deploymentPath = runOptions.deploymentPath;
+  const yamlFileName = runOptions.yamlFileName;
+  const storageBucket = runOptions.storageBucket;
 
   // Move YAML.
   const yamlPath = path.join(deploymentPath, yamlFileName);
-  if (copyYaml) {
-    task.cp(path.join(sourceFolder, yamlFileName), deploymentPath);
+  if (runOptions.copyYaml) {
+    const yamlSource = runOptions.yamlSource;
+    const appendedSource = path.join(yamlSource, yamlFileName);
+    if (yamlSource.endsWith(yamlFileName) && !task.exist(appendedSource)) {
+      task.cp(yamlSource, deploymentPath);
+    } else {
+      task.cp(appendedSource, deploymentPath);
+    }
   }
 
   // Set version
-  const version = versionInput || isoNowString();
-
-  // Set up the key file from the certificate parameter of the service
-  // endpoint.
-  const endpoint = new Endpoint(endpointId);
+  const version = runOptions.versionInput || isoNowString();
 
+  const endpoint = runOptions.endpoint;
   // Set gcloud arguments.
   const projectArg = endpoint.projectParam;
   const credentialArg = Endpoint.credentialParam;
 
   const gcloud: ToolRunner =
-      task.tool(gcloudPath)
-          .line('beta app deploy --quiet --verbosity=info')
-          .arg([ `"${yamlPath}"`, credentialArg, projectArg ])
-          .argIf(version, `--version="${version}"`)
-          .argIf(storageBucket, `--bucket="${storageBucket}"`)
-          .argIf(promote, '--promote')
-          .argIf(!promote, '--no-promote')
-          .argIf(promote && stopPrevious, '--stop-previous-version')
-          .argIf(promote && !stopPrevious, '--no-stop-previous-version');
+    task.tool(gcloudPath)
+      .line('beta app deploy --quiet --verbosity=info')
```

Comment:
Nit: You can remove the "beta" portion now. We're fully supported and GA.

Reply:
Tracking this in issue #39.